Code/Framework/AzFramework/Tests/Utils/Utils.cpp

```diff
@@ -8,72 +8,100 @@
 
 #include "Utils.h"
 #include <AzCore/IO/Path/Path.h>
+#include <AzCore/std/functional.h>
 #include <AzCore/StringFunc/StringFunc.h>
 
-UnitTest::ScopedTemporaryDirectory::ScopedTemporaryDirectory()
+namespace UnitTest
 {
-    constexpr int MaxAttempts = 255;
+    void DeleteFolderRecursive(const AZ::IO::PathView& path)
```

Comment:
nit: If this is a 'view' should it just be passed by value and not const ref?

Reply:
Well it is view, but it is made up of a 16 byte `AZStd::string_view` and a `1 byte` path separator character.
So it is really 24 bytes in length.

Reply:
Right but the received wisdom is to pass `std::string_view` by value and not by const ref which is what I was getting at, should this type be treated the same way?

Reply:
Well once the size of a class is above 16 bytes, passing by reference is normally more efficient.